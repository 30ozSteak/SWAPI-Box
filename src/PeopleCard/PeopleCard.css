.people-card-literal {
  border: 1px solid white;
  background-color: red;
  width: 20rem;
  height: 25rem;
  border-radius: 5px;
  position: relative;
  z-index: 4;
}
