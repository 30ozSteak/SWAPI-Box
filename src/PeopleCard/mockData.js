const people = {
    "count": 87, 
    "next": "https://swapi.co/api/people/?page=2", 
    "previous": null, 
    "results": [
        {
            "name": "Luke Skywalker", 
            "height": "172", 
            "mass": "77", 
            "hair_color": "blond", 
            "skin_color": "fair", 
            "eye_color": "blue", 
            "birth_year": "19BBY", 
            "gender": "male", 
            "homeworld": "https://swapi.co/api/planets/1/", 
            "films": [
                "https://swapi.co/api/films/2/", 
                "https://swapi.co/api/films/6/", 
                "https://swapi.co/api/films/3/", 
                "https://swapi.co/api/films/1/", 
                "https://swapi.co/api/films/7/"
            ], 
            "species": [
                "https://swapi.co/api/species/1/"
            ], 
            "vehicles": [
                "https://swapi.co/api/vehicles/14/", 
                "https://swapi.co/api/vehicles/30/"
            ], 
            "starships": [
                "https://swapi.co/api/starships/12/", 
                "https://swapi.co/api/starships/22/"
            ], 
            "created": "2014-12-09T13:50:51.644000Z", 
            "edited": "2014-12-20T21:17:56.891000Z", 
            "url": "https://swapi.co/api/people/1/"
        }, 
        {
            "name": "C-3PO", 
            "height": "167", 
            "mass": "75", 
            "hair_color": "n/a", 
            "skin_color": "gold", 
            "eye_color": "yellow", 
            "birth_year": "112BBY", 
            "gender": "n/a", 
            "homeworld": "https://swapi.co/api/planets/1/", 
            "films": [
                "https://swapi.co/api/films/2/", 
                "https://swapi.co/api/films/5/", 
                "https://swapi.co/api/films/4/", 
                "https://swapi.co/api/films/6/", 
                "https://swapi.co/api/films/3/", 
                "https://swapi.co/api/films/1/"
            ], 
            "species": [
                "https://swapi.co/api/species/2/"
            ], 
            "vehicles": [], 
            "starships": [], 
            "created": "2014-12-10T15:10:51.357000Z", 
            "edited": "2014-12-20T21:17:50.309000Z", 
            "url": "https://swapi.co/api/people/2/"
        }
    ]
}

const planets = [
        [{
            "name": "Alderaan", 
            "rotation_period": "24", 
            "orbital_period": "364", 
            "diameter": "12500", 
            "climate": "temperate", 
            "gravity": "1 standard", 
            "terrain": "grasslands, mountains", 
            "surface_water": "40", 
            "population": "2000000000", 
            "residents": [
                "https://swapi.co/api/people/5/", 
                "https://swapi.co/api/people/68/", 
                "https://swapi.co/api/people/81/"
            ], 
            "films": [
                "https://swapi.co/api/films/6/", 
                "https://swapi.co/api/films/1/"
            ], 
            "created": "2014-12-10T11:35:48.479000Z", 
            "edited": "2014-12-20T20:58:18.420000Z", 
            "url": "https://swapi.co/api/planets/2/"
        }, 
        {
            "name": "Yavin IV", 
            "rotation_period": "24", 
            "orbital_period": "4818", 
            "diameter": "10200", 
            "climate": "temperate, tropical", 
            "gravity": "1 standard", 
            "terrain": "jungle, rainforests", 
            "surface_water": "8", 
            "population": "1000", 
            "residents": [], 
            "films": [
                "https://swapi.co/api/films/1/"
            ], 
            "created": "2014-12-10T11:37:19.144000Z", 
            "edited": "2014-12-20T20:58:18.421000Z", 
            "url": "https://swapi.co/api/planets/3/"
        }]
    ]

export {
    people
}

export {
    planets
}