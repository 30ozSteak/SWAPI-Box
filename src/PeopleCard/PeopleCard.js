import React, { Component } from "react";
import "./PeopleCard.css";
import Images from "../Images.json";

const PeopleCard = ({ people, planets, species, handleFavorites }) => {
  let characterArray = [];

  let answer = people.results.map(person => {
    let personName = person.name;
    let personWorld = person.homeworld;
    planets.map(planet => {
      planet.map(index => {
        let indexPosition = planet.indexOf(index);
        if (personWorld === planet[indexPosition].url) {
          for (let i in species) {
            if (species[i].people.includes(person.url)) {
              let peopleObject = {
                name: personName,
                location: planet[indexPosition].name,
                species: species[i].name,
                population: planet[indexPosition].population
              };
              characterArray.push(peopleObject);
            }
          }
        }
      });
    });
  });

  let randomizedKey =() => {
    return Math.floor(Math.random() * (10000 - 1 + 1)) + 1 + Date.now() + Math.random()
  } 

console.log(characterArray)
  const peopleStats = characterArray.map(data => {
    return (
      <div key={data.name} className="card-literal">
        <section className="image">
          <img
            className="image-literal"
            src={Images[data.name]}
            alt={data.name}
          />
          <div className="fave" onClick={() => handleFavorites(data)} />
          <section className="text-field">
            <h2>
              {data.name}
            </h2>
            <h4>
              Species: {data.species}
            </h4>
            <div
              className="caret"
            />
            <h4>
              HomeWorld: {data.location}
            </h4>
            <h4>
              Population: {data.population}
            </h4>
          </section>
        </section>
      </div>
    );
  });

  return <div>{peopleStats}</div>;
};

export default PeopleCard;

