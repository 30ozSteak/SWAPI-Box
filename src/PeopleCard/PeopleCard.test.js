import React from "react";
import PeopleCard from "./PeopleCard";
import { shallow } from "enzyme";

describe.skip("PeopleCard", () => {

  it('should exist', () => {
    const people = [{results: [{name: "Luke"}, {name: 'Darth Vador'}]}]
    const mockPlanets = [{homeworld: "Moon"}]
    const mockSpecies = [{species: "driod"}]
    const wrapper = shallow(<PeopleCard people={people} planets={mockPlanets} species={mockSpecies} />);
    expect(wrapper).toBeDefined();
  });


  it("should match the screenshot", () => {
    const people = [{results: [{name: "Luke"}, {name: 'Darth Vador'}]}]

  console.log(people.results)

    const mockPlanets = [{homeworld: "Moon"}]
    const mockSpecies = [{species: "driod"}]
    const characterArray = [people, mockPlanets, mockSpecies] 

    const wrapper = shallow(<PeopleCard people={people} planets={mockPlanets} species={mockSpecies} />);

    expect(wrapper).toMatchSnapshot();
  });
})
