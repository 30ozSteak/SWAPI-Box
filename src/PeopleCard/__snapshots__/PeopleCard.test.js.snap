// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCard should match the screenshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PeopleCard
    people={
      Object {
        "count": 87,
        "next": "https://swapi.co/api/people/?page=2",
        "previous": null,
        "results": Array [
          Object {
            "birth_year": "19BBY",
            "created": "2014-12-09T13:50:51.644000Z",
            "edited": "2014-12-20T21:17:56.891000Z",
            "eye_color": "blue",
            "films": Array [
              "https://swapi.co/api/films/2/",
              "https://swapi.co/api/films/6/",
              "https://swapi.co/api/films/3/",
              "https://swapi.co/api/films/1/",
              "https://swapi.co/api/films/7/",
            ],
            "gender": "male",
            "hair_color": "blond",
            "height": "172",
            "homeworld": "https://swapi.co/api/planets/1/",
            "mass": "77",
            "name": "Luke Skywalker",
            "skin_color": "fair",
            "species": Array [
              "https://swapi.co/api/species/1/",
            ],
            "starships": Array [
              "https://swapi.co/api/starships/12/",
              "https://swapi.co/api/starships/22/",
            ],
            "url": "https://swapi.co/api/people/1/",
            "vehicles": Array [
              "https://swapi.co/api/vehicles/14/",
              "https://swapi.co/api/vehicles/30/",
            ],
          },
          Object {
            "birth_year": "112BBY",
            "created": "2014-12-10T15:10:51.357000Z",
            "edited": "2014-12-20T21:17:50.309000Z",
            "eye_color": "yellow",
            "films": Array [
              "https://swapi.co/api/films/2/",
              "https://swapi.co/api/films/5/",
              "https://swapi.co/api/films/4/",
              "https://swapi.co/api/films/6/",
              "https://swapi.co/api/films/3/",
              "https://swapi.co/api/films/1/",
            ],
            "gender": "n/a",
            "hair_color": "n/a",
            "height": "167",
            "homeworld": "https://swapi.co/api/planets/1/",
            "mass": "75",
            "name": "C-3PO",
            "skin_color": "gold",
            "species": Array [
              "https://swapi.co/api/species/2/",
            ],
            "starships": Array [],
            "url": "https://swapi.co/api/people/2/",
            "vehicles": Array [],
          },
        ],
      }
    }
    planets={
      Array [
        Array [
          Object {
            "climate": "temperate",
            "created": "2014-12-10T11:35:48.479000Z",
            "diameter": "12500",
            "edited": "2014-12-20T20:58:18.420000Z",
            "films": Array [
              "https://swapi.co/api/films/6/",
              "https://swapi.co/api/films/1/",
            ],
            "gravity": "1 standard",
            "name": "Alderaan",
            "orbital_period": "364",
            "population": "2000000000",
            "residents": Array [
              "https://swapi.co/api/people/5/",
              "https://swapi.co/api/people/68/",
              "https://swapi.co/api/people/81/",
            ],
            "rotation_period": "24",
            "surface_water": "40",
            "terrain": "grasslands, mountains",
            "url": "https://swapi.co/api/planets/2/",
          },
          Object {
            "climate": "temperate, tropical",
            "created": "2014-12-10T11:37:19.144000Z",
            "diameter": "10200",
            "edited": "2014-12-20T20:58:18.421000Z",
            "films": Array [
              "https://swapi.co/api/films/1/",
            ],
            "gravity": "1 standard",
            "name": "Yavin IV",
            "orbital_period": "4818",
            "population": "1000",
            "residents": Array [],
            "rotation_period": "24",
            "surface_water": "8",
            "terrain": "jungle, rainforests",
            "url": "https://swapi.co/api/planets/3/",
          },
        ],
      ]
    }
    species={
      Array [
        Object {
          "species": "driod",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
